src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      const newState = {};
+
+      for (const key in currentState) {
+        if (!action.keysToRemove.includes(key)) {
+          newState[key] = currentState[key];
+        }
+      }
+      currentState = newState;
+    }
+
+    states.push(currentState);
```

Comment:
The action of pushing the `currentState` into the `states` array is repeated in all cases. To adhere to the DRY principle, perform this action once after the switch statement.